src/program/lwaftr/run/run.lua

```diff
@@ -20,30 +21,27 @@ end
 local function migrate_device_on_config(config, v4, v6)
    -- Validate there is only one instance, otherwise the option is ambiguous.
    local device, instance
-   for pci_addr, inst in pairs(config.softwire_config.instance) do
-      assert(device == nil, "Unable to migrate config to '"..pci_addr.."' as"..
-                             "there are multiple instances configured.")
-      device, instance = pci_addr, inst
+   for k, v in pairs(config.softwire_config.instance) do
+      assert(device == nil,
+             "Unable to specialize config for specified NIC(s) as"..
+                "there are multiple instances configured.")
+      device, instance = k, v
    end
-
-   local migrated = config
-   migrated.softwire_config.instance = lib.deepcopy(
-      migrated.softwire_config.instance
-   )
-   local instances = migrated.softwire_config.instance
+   assert(device ~= nil,
+          "Unable to specialize config for specified NIC(s) as"..
+             "there are no instances configured.")
 
    if v4 and v4 ~= device then
       print("Migrating instance '"..device.."' to '"..v4.."'")
-      instances[v4] = instances[device]
-      instances[device] = nil
+      config.softwire_config.instance[v4] = instance
+      config.softwire_config.instance[device] = nil
```

Comment:
I guess it's okay but wouldn't this mutate the config passed in rather than producing a new one which will be returned leaving the previous unchanged?

Reply:
Yes, this does mutate the config passed in.  However since it's passed in from the same file and the old config isn't used anywhere else (either before or after), I think it's OK to frob it in place.  OK by you?

Reply:
Sure, I try to avoid doing that but it's more of a precaution and conceptual thing, I have a preference for it returning a new config but it can remain as is.